library/core/src/ops/control_flow.rs

```diff
@@ -0,0 +1,67 @@
+use crate::ops::Try;
+
+/// Used to make try_fold closures more like normal loops
+#[unstable(feature="control_flow_enum", reason="new API", issue="75744")]
+#[derive(Debug, Clone, Copy, PartialEq)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

`Hash` and especially `Eq` seem handy, although they are not needed to replace `LoopState` and are less essential for `ControlFlow` than for `Option`.

Reply:
Yes, that seems sensible! I don't know when you'd want to hash this but I suppose there's no reason to prevent it.